docs/pip/packages.md

```diff
@@ -107,6 +107,18 @@ Install from a `pyproject.toml` file with all optional dependencies enabled:
 $ uv pip install -r pyproject.toml --all-extras
 ```
 
+To install dependency groups in the current project directory's pyproject.toml:
+
+```console
+$ uv pip install --group foo
+```
+
+To install dependency groups for an arbitrary pyproject.toml:
+
+```console
+$ uv pip install --group some/path/pyproject.toml:foo
+```
```

Comment:
We need to talk about how `--group` does not apply to other sources, like `-e .` etc. here. Maybe under an `!!! note` admonition? We could track this separately, but this is an important thing to teach in the docs.